debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/YugabyteReplicaIdentityIT.java

```diff
@@ -0,0 +1,350 @@
+    SourceRecord deleteRecord = records.get(1);
+
+    YBVerifyRecord.isValidInsert(insertRecord, PK_FIELD, 1);
+    YBVerifyRecord.isValidDelete(deleteRecord, PK_FIELD, 1);
+
+    Struct deleteRecordValue = (Struct) deleteRecord.value();
+    assertThat(deleteRecordValue.get(Envelope.FieldName.AFTER)).isNull();
+    assertThat(deleteRecordValue.get(Envelope.FieldName.BEFORE)).isNotNull();
+
+    // Before field will have entries for all the columns.
+    assertThat(deleteRecordValue.getStruct(Envelope.FieldName.BEFORE).getStruct("pk").getInt32("value")).isEqualTo(1);
+    assertThat(deleteRecordValue.getStruct(Envelope.FieldName.BEFORE).getStruct("aa").getInt32("value")).isEqualTo(22);
+    assertThat(deleteRecordValue.getStruct(Envelope.FieldName.BEFORE).getStruct("bb").getString("value")).isEqualTo("random text value");
+  }
+
+  @Test
+  public void shouldHaveBeforeImageForDeletesForReplicaIdentityDefault() throws Exception {
+    TestHelper.execute("ALTER TABLE s2.a REPLICA IDENTITY DEFAULT;");
+
+    Configuration config = TestHelper.defaultConfig()
+                             .with(PostgresConnectorConfig.SNAPSHOT_MODE, PostgresConnectorConfig.SnapshotMode.NEVER.getValue())
+                             .with(PostgresConnectorConfig.DROP_SLOT_ON_STOP, Boolean.TRUE)
+                             .build();
+    start(PostgresConnector.class, config);
+    assertConnectorIsRunning();
+
+    // YB Note: Added a wait for replication slot to be active.
+    TestHelper.waitFor(Duration.ofSeconds(10));
+
+    waitForAvailableRecords(10_000, TimeUnit.MILLISECONDS);
+    // there shouldn't be any snapshot records
+    assertNoRecordsToConsume();
+
+    // insert and verify 2 new records
+    TestHelper.execute("INSERT INTO s2.a VALUES (1, 22, 'random text value');");
+    TestHelper.execute("DELETE FROM s2.a WHERE pk = 1;");
+
+    SourceRecords actualRecords = consumeRecordsByTopic(2);
+    List<SourceRecord> records = actualRecords.recordsForTopic(topicName("s2.a"));
+
+    SourceRecord insertRecord = records.get(0);
+    SourceRecord deleteRecord = records.get(1);
+
+    YBVerifyRecord.isValidInsert(insertRecord, PK_FIELD, 1);
+    YBVerifyRecord.isValidDelete(deleteRecord, PK_FIELD, 1);
+
+    Struct deleteRecordValue = (Struct) deleteRecord.value();
+    assertThat(deleteRecordValue.get(Envelope.FieldName.AFTER)).isNull();
+    assertThat(deleteRecordValue.get(Envelope.FieldName.BEFORE)).isNotNull();
+
+    // Before field will have entries only for the primary key columns.
+    assertThat(deleteRecordValue.getStruct(Envelope.FieldName.BEFORE).getStruct("pk").getInt32("value")).isEqualTo(1);
+    assertThat(deleteRecordValue.getStruct(Envelope.FieldName.BEFORE).getStruct("aa").getInt32("value")).isNull();
+    assertThat(deleteRecordValue.getStruct(Envelope.FieldName.BEFORE).getStruct("bb").getString("value")).isNull();
+  }
+
+  @Test
+  public void shouldHaveBeforeImageForDeletesForReplicaIdentityChange() throws Exception {
```

Comment:
Can we combine this with `shouldProduceEventsWithValuesForChangedColumnWithReplicaIdentityChange`.